src/services/contract_wrappers/fees_wrapper.js

```diff
@@ -18,4 +19,11 @@ export default class FeesWrapper extends ContractWrapper {
     const contract = await this.contract();
     return contract.methods.getFeeForUpload(storagePeriods).call();
   }
+
+  async ensureEnoughFunds(fee) {
+    const balance = new BN(await this.web3.eth.getBalance(this.contractManager.defaultAddress()));
+    if (balance.lt(new BN(fee))) {
+      throw new Error(`Insufficient funds: required: ${fee} and got ${balance}`);
+    }
+  }
```

Comment:
IMO too much logic for a wrapper. Let wrapper be wrapper and move it out somewhere else. Maybe to `uploadRepository`? Also I'd make direct call to fees (obvsly through `feesWrapper`) to get current fee, instead of passing it to method.

Reply:
i wish i could but we use wrappers to separate web3 from rest of the world, so it's either this or put a getBalance method to some wrapper